cpu/stm32_common/stm32_mem_lengths.mk

```diff
@@ -0,0 +1,304 @@
+      endif
+    else ifeq ($(STM32_MODEL),303)
+      ifneq (, $(filter $(STM32_ROMSIZE), 6 8))
+        RAM_LEN = 16K
+      else ifneq (, $(filter $(STM32_ROMSIZE), B))
+        RAM_LEN = 40K
+      else ifneq (, $(filter $(STM32_ROMSIZE), C))
+        RAM_LEN = 48K
+      else ifneq (, $(filter $(STM32_ROMSIZE), D E))
+        RAM_LEN = 80K
+      endif
+    else ifeq ($(STM32_MODEL3),4)
+      RAM_LEN = 16K
+    else ifeq ($(STM32_MODEL),373)
+      RAM_LEN = 32K
+    else ifeq ($(STM32_MODEL3),8)
+      ifneq (, $(filter $(STM32_MODEL2), 1 2))
+        RAM_LEN = 16K
+      else ifneq (, $(filter $(STM32_MODEL2), 5))
+        RAM_LEN = 48K
+      else ifneq (, $(filter $(STM32_MODEL2), 7))
+        RAM_LEN = 32K
+      else ifneq (, $(filter $(STM32_MODEL2), 9))
+        RAM_LEN = 80K
+      endif
+  endif
+  else ifeq ($(STM32_FAMILY),4)
+    ifeq ($(STM32_MODEL),401)
+      ifneq (, $(filter $(STM32_ROMSIZE), B C))
+        RAM_LEN = 64K
+      else ifneq (, $(filter $(STM32_ROMSIZE), D E))
+        RAM_LEN = 96K
+      endif
+    else ifneq (, $(filter $(STM32_MODEL), 405 407))
+      RAM_LEN = 192K
+    else ifeq ($(STM32_MODEL),410)
+      RAM_LEN = 32K
+    else ifneq (, $(filter $(STM32_MODEL), 411 446))
+      RAM_LEN = 128K
+    else ifneq (, $(filter $(STM32_MODEL), 412 427 429 437 439))
+      RAM_LEN = 256K
+    else ifneq (, $(filter $(STM32_MODEL), 413 423))
+      RAM_LEN = 320K
+    else ifneq (, $(filter $(STM32_MODEL), 415 417))
+      RAM_LEN = 192K
+    else ifneq (, $(filter $(STM32_MODEL), 469 479))
+      RAM_LEN = 384K
+    endif
+  else ifeq ($(STM32_FAMILY),7)
+    ifneq (, $(filter $(STM32_MODEL2), 2 3))
+      RAM_LEN = 256K
+    else ifneq (, $(filter $(STM32_MODEL2), 4 5))
+      RAM_LEN = 320K
+    else ifneq (, $(filter $(STM32_MODEL2), 6 7))
+      RAM_LEN = 512K
+    endif
+  endif
+else ifeq ($(STM32_TYPE), L)
+    ifeq ($(STM32_FAMILY),0)
+        ifneq (, $(filter $(STM32_MODEL2), 1 2))
+            RAM_LEN = 2K
+        else ifneq (, $(filter $(STM32_MODEL2), 3 4 5 6))
+            RAM_LEN = 8K
+        else ifneq (, $(filter $(STM32_MODEL2), 7 8))
+            RAM_LEN = 20K
+        endif
+    else ifeq ($(STM32_FAMILY),1)
+      ifeq ($(STM32_MODEL), 100)
+        ifneq (, $(filter $(STM32_ROMSIZE), 6))
+          RAM_LEN = 4K
+        else ifneq (, $(filter $(STM32_ROMSIZE), 8))
+          RAM_LEN = 8K
+        else ifneq (, $(filter $(STM32_ROMSIZE), B C))
+          RAM_LEN = 16K
+        endif
+      else ifneq (, $(filter $(STM32_MODEL), 151 152))
+        ifneq (, $(filter $(STM32_PINCOUNT), C Q U V Z))
+          ifneq (, $(filter $(STM32_ROMSIZE), 6))
+            RAM_LEN = 16K
+          else ifneq (, $(filter $(STM32_ROMSIZE), 8 B C))
+            RAM_LEN = 32K
+          else ifneq (, $(filter $(STM32_ROMSIZE), D))
+            RAM_LEN = 48K
+          else ifneq (, $(filter $(STM32_ROMSIZE), E))
+            RAM_LEN = 80K
+          else ifneq (, $(filter $(STM32_ROMSIZE), DT6X))
```

Comment:
`STM32_ROMSIZE` is supposed to be one character, I think this should not work

Reply:
Actually I put them there more for information rather than for compatibility. This models are kind of rare (some of them are part of just a first lot of such a model), thus I don't know if it worths the pain to modify all the CPUs to force the "long" name, which would make this weird models compatible with the script... Do you have an easier suggestion to make this work?

Reply:
I think it is better to leave them out for now if we don't have any boards with those models. The regex will need to be adapted if you want to catch this model and I can't see the benefit now.

Reply:
I'm OK with that